### PR TITLE
Support TimeZones 1.0

### PR DESCRIPTION
**Pull request by omus**
Closes #221 

Updated Project.toml to support TimeZones 1.0. I also refactored some of the code relating time time zones to clean things up a bit. I'll note that `localzone()` is kinda slow so using UTC by default  makes performance a bit better.

### CODE REVIEW

File: test/automerge-integration-utils.jl [hunk 1]
```diff
@@ -199,7 +193,7 @@ end
 
 function _generate_branch_name(name::AbstractString)
     sleep(0.1)
-    _now = now_localzone()
+    _now_utc = now(tz"UTC")
     _now_utc_string = utc_to_string(_now)
```
Comment (omus) [inline test/automerge-integration-utils.jl]: I messed this line up

**Comment by omus** [comment]
I'm not sure what the issue is with the automerge tests. Could use some guidance there.

**Comment by omus** [comment]
Bump @DilumAluthge

File: Project.toml [hunk 2]
```diff
@@ -21,17 +21,11 @@ GitHub = "5.1.1"
 HTTP = "0.8"
 JSON = "0.19, 0.20, 0.21"
 RegistryTools = "1.2"
-TimeZones = "0.10"
+TimeZones = "0.10.3, 0.11, 1"
```
Comment (DilumAluthge) [inline Project.toml:24]: I'd rather drop support for pre-1.0 TimeZones.

```suggestion
TimeZones = "1"
```

Comment (omus) [reply Project.toml:24]: Works for me

**Comment by omus** [comment]
Rebased and resolve conflicts

**Comment by omus** [comment]
Fixed a mistake I made in `string_to_utc`

**Comment by DilumAluthge** [comment]
Can you squash? Then I'll try running the integration tests.

**Comment by omus** [comment]
> Can you squash? Then I'll try running the integration tests.

Done

**Pull request merged**